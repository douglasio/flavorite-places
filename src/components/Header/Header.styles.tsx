import styled from 'styled-components'
import { Anchor } from 'grommet'
// import theme from 'styles/theme'

export const Lockup = styled(Anchor)`
	font-family: 'Barrio', Arial, Helvetica, sans-serif;
	font-size: 2rem;

	svg {
		min-width: 3rem;
	}

	span {
		white-space: nowrap;
	}

	.accent {
		filter: brightness(150%);
	}
`

export const NavLink = styled(Anchor)`
	text-transform: capitalize;
`
