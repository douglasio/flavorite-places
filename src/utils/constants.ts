export const siteTitle = 'flavorite places'
