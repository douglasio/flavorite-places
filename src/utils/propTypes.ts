import PropTypes from 'prop-types'

export const menuItemsPropTypes = PropTypes.arrayOf(PropTypes.shape({ label: PropTypes.string, href: PropTypes.string }))
