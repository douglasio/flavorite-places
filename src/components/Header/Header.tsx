//Dependencies
import React from 'react'
import PropTypes from 'prop-types'
import { menuItemsPropTypes } from 'utils/propTypes'
import { Box, Header as GrommetHeader, Menu, Nav, ResponsiveContext } from 'grommet'

//Components
import * as Styled from './Header.styles'
import Logo from 'svg/Logo'

//Styles

type MenuItemTypes = {
	label: string
	href: string
}

type LayoutProps = {
	menuItems: Array<MenuItemTypes>
	siteTitle: string
}

const Header = ({ siteTitle, menuItems }: LayoutProps) => {
	// const size = React.useContext(ResponsiveContext)

	return (
		<GrommetHeader justify='between'>
			<Box direction='row' align='center' gap='medium' pad={{ vertical: 'small', horizontal: 'large' }}>
				<Styled.Lockup
					title={siteTitle}
					color='brand'
					icon={<Logo foreground='#bb72ff' />}
					href={'/'}
					label={
						<>
							<span className='accent'>flavor</span>ite places
						</>
					}
					size='large'
				></Styled.Lockup>
			</Box>
			<Box direction='row' align='center' gap='medium' pad={{ vertical: 'small', horizontal: 'large' }}>
				<ResponsiveContext.Consumer>
					{(size) =>
						size === 'small' ? (
							<Menu a11yTitle='Navigation menu' dropProps={{ align: { top: 'bottom', right: 'right' } }} items={menuItems}></Menu>
						) : (
							<Nav direction='row' gap='medium' pad={{ vertical: 'small', horizontal: 'large' }}>
								{menuItems.map((menuItem: MenuItemTypes, i) => {
									return <Styled.NavLink key={`${menuItem.label}${i}`} {...menuItem} size='medium' />
								})}
							</Nav>
						)
					}
				</ResponsiveContext.Consumer>
			</Box>
		</GrommetHeader>
	)
}

Header.propTypes = {
	menuItems: menuItemsPropTypes,
	siteTitle: PropTypes.string.isRequired,
}

export default Header
