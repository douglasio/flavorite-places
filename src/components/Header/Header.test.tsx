import React from 'react'
import { StaticQuery } from 'gatsby'
import { render, screen } from '@testing-library/react'
import { siteTitle } from 'utils/constants'

import Header from './Header'

const mockMenuItems = [
	{ label: 'Link 1', href: '/' },
	{ label: 'Link 2', href: '/' },
]

const TestHeader = () => <Header data-testid='testHeader' menuItems={mockMenuItems} siteTitle={siteTitle} />

test('displays all menu items', () => {
	render(<TestHeader />)
	mockMenuItems.forEach((item) => {
		expect(screen.getByText(item.label)).toBeInTheDocument()
	})
})
