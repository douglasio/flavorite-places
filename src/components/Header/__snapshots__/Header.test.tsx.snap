// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="StyledBox-sc-13pk1d4-0 cnREfB"
>
  <div
    className="StyledBox-sc-13pk1d4-0 heSymd"
  >
    <a
      aria-label="Site Title"
      className="StyledAnchor-sc-1rp7lwl-0 DrOgy sc-bdvvtL EAfBV"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
      size="large"
    >
      <span
        className="StyledBox-sc-13pk1d4-0 deExzV"
        style={
          Object {
            "display": "inline-flex",
          }
        }
      >
        <svg
          viewBox="0 0 99 117.647"
        >
          <defs>
            <clipPath
              id="clipPath"
            >
              <ellipse
                cx="49.5"
                cy="49"
                data-name="Ellipse 13"
                fill="rgba(0,0,0,0.26)"
                id="Ellipse_13"
                rx="49.5"
                ry="49"
                transform="translate(0 -0.359)"
              />
            </clipPath>
          </defs>
          <g
            data-name="Group 389"
            id="Group_389"
            transform="translate(4607 7017.595)"
          >
            <g
              clipPath="url(#clipPath)"
              data-name="Mask Group 13"
              id="Mask_Group_13"
              transform="translate(-4607 -7005.641)"
            >
              <rect
                data-name="Rectangle 213"
                fill="rgba(223,223,223,0.44)"
                height="111"
                id="Rectangle_213"
                transform="translate(-22 -3.359)"
                width="131"
              />
              <path
                d="M-194.434,5894.789c-14.282-14.735,12.82-11.4,10.97-16.139l-.1-.153c-9.157-.595-16.435-9.987-16.435-21.487,0-11.882,7.772-21.515,17.358-21.515,5.417,0,14.479,5.011,18.979,10.453-15.974,5.395,4.138,8.532-1.619,11.063-6.194,2.725-15.878.74-17.359,5.228-8.472-.707-3.367,6.783-.547,11.845a13.3,13.3,0,0,1,4.315,3.729c4.96,1.261,5.558,5.469,8.2,8.122,4.343,1.14-2.574,10.01-2.574,14.829-5.951,7.9-10.7,3.511-12.545,13.688-.353,1.942-.886,2.769-1.518,2.77C-189.99,5917.222-194.434,5902.375-194.434,5894.789Zm63.288-7.018c-.578-1.227-3.319-.563-6.626.1-3.473.7-7.574,1.395-10.454-.1-4.413-2.354-7.13-15.417-3.656-16.149.448-2.061,3.244-6.452,7.039-7.39,4.135-1.021,7.124-1.862,7.165-3.634.064-2.845-3.14-1.9-6.31-.959-3.072.916-6.111,1.829-6.111-.707-8.958-.4-4.953-2.312-3.9-5.036,4.025.5,4.138-.969,3.841-2.443-.315-1.575-1.1-3.151,1.933-2.308,2.8-1.416,3.314-8.805,6.872-10.621,2.876.059,5.353-1.942,7.66-3.942,2.834-2.457,5.414-4.913,8.164-3.555,10.756,5.311,18.842,26.963,18.842,38.705,0,17.117,9.9,46.544-7.216,46.544C-119.159,5916.277-127.411,5895.7-131.146,5887.771Z"
                data-name="Union 3"
                fill="rgba(255,255,255,0.47)"
                id="Union_3"
                transform="translate(206.439 -5835.339)"
              />
            </g>
            <g
              data-name="Group 351"
              id="Group_351"
              transform="translate(-4581.736 -7017.595)"
            >
              <g
                data-name="Group 350"
                id="Group_350"
                transform="translate(0)"
              >
                <path
                  d="M-18.928,0H4.393L-5.133,13.625l1.256,61.538h-6.654l2.175-61.538Z"
                  data-name="Path 82"
                  fill="#fff"
                  id="Path_82"
                  transform="translate(18.928 42.484)"
                />
                <path
                  d="M3.73,4.061,5.207,2.373,7.285,0V42.484H3.73Z"
                  data-name="Path 79"
                  fill="#fff"
                  id="Path_79"
                  transform="translate(-3.73)"
                />
                <path
                  d="M6.928,4.061,5.6,2.373,3.73,0V42.484h3.2Z"
                  data-name="Path 80"
                  fill="#fff"
                  id="Path_80"
                  transform="translate(16.392)"
                />
                <path
                  d="M0-4.9H3.949V37.587H0Z"
                  data-name="Path 81"
                  fill="#fff"
                  id="Path_81"
                  transform="translate(9.929 5.433)"
                />
              </g>
              <g
                data-name="Group 349"
                id="Group_349"
                transform="translate(28.93 12.882)"
              >
                <path
                  d="M0,0H5.353L2.161,91.529H0Z"
                  data-name="Path 83"
                  fill="#fff"
                  id="Path_83"
                />
                <path
                  d="M0,0H5.353L2.161,91.529H0Z"
                  data-name="Path 84"
                  fill="#fff"
                  id="Path_84"
                  transform="translate(18.693 9.33) rotate(10)"
                />
              </g>
            </g>
          </g>
        </svg>
        <span
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iUEnJr"
        />
        <span
          className="accent"
        >
          flavor
        </span>
        ite places
      </span>
    </a>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 heSymd"
  >
    <nav
      className="StyledBox-sc-13pk1d4-0 bNzMNA"
    >
      <a
        className="StyledAnchor-sc-1rp7lwl-0 OIdWf sc-gsDKAQ eeCFXs"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
        size="medium"
      >
        Test
      </a>
    </nav>
  </div>
</header>
`;
